rust/src/adapter/datadog_formatter/mod.rs

```diff
@@ -0,0 +1,83 @@
+use anyhow::Result;
+use std::{time::SystemTime, collections::HashMap};
+
+use serde::Serialize;
+
+use super::{datadog::DatadogConfig, new_span_id};
+
+pub struct DatadogFormatter {
+    pub traces: Vec<Trace>
+}
+
+pub type Trace = Vec<Span>;
+
+impl DatadogFormatter {
+    pub fn new() -> DatadogFormatter {
+        DatadogFormatter {
+            traces: Vec::new(),
+        }
+    }
+
+    pub fn add_trace(&mut self, trace: Trace) {
+        self.traces.push(trace)
+    }
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize)]
+pub struct Span {
+    pub trace_id: u64,
+    pub span_id: u64,
+    pub parent_id: u64,
+    pub name: String,
+    pub start: u64,
+    pub duration: u64,
+    pub resource: String,
+    pub error: u8,
+    pub meta: HashMap<String, String>,
+    pub metrics: HashMap<String, f64>,
+    pub service: String,
+    #[serde(rename = "type")]
+    pub typ: Option<String>,
+}
+
+impl Span {
+    pub fn new(
+        config: DatadogConfig,
+        trace_id: u64,
+        parent_id: Option<u64>,
+        name: String,
+        start_time: SystemTime,
+        end_time: SystemTime,
+    ) -> Result<Span> {
+        let span_id = new_span_id().into();
+
+        let p_id = match parent_id {
+            Some(id) => id,
+            None => new_span_id().into(),
```

Comment:
So this means the root span will have a random parent_id? Is it supposed to point to nothing instead of being blank?

Reply:
yes, good eye. I think I ported this incorrectly. let me fix it. I believe it should be empty.